docs/rpc.md

```diff
@@ -86,15 +91,22 @@ Note that depending on the concrete implementation you use to run the server, on
 In order to implement the service, you have to define the behavior of each method by means of a *handler*. You can use Haskell types for your handlers, given that you had previously declared that they can be mapped back and forth the schema types using `HasSchema`. For example, the following is a handler for the `SayHello` method in `Greeter`:
 
 ```haskell
-sayHello :: HelloRequest -> ServerErrorIO HelloResponse
+sayHello :: (MonadServer m) => HelloRequest -> m HelloResponse
 sayHello (HelloRequest nm) = return (HelloResponse ("hi, " <> nm))
 ```
 
-Since you can declare more than one method in a service, you need to join then into a `Server`. You do so by using `(:<|>:)` between each handler and ending the sequence with `H0`. In addition to the name of the service, `Server` has an additional parameter which records the types of the handlers. Since that list may become quite long, we can ask GHC to write it for us by using the `PartialTypeSignatures` extension and writing an underscore `_` in that position. One final observation is that in the code below we are using `ServerIO`, which is an instance of `Server` which allows running `IO` operations.
+Notice the use of `MonadServer` in this case. This gives us the ability to:
+
+* Run arbitrary `IO` actions by using `liftIO`,
+* Return an error code by calling `serverError`.
+
+Being polymorphic here allows us to run the same server in multiple back-ends. Furthermore, by enlargint the set of abilities required for our monad `m`, we can [integrate with other libraries](transformer.md), including logging and resource pools.
+
+Since you can declare more than one method in a service, you need to join then into a `Server`. You do so by using `(:<|>:)` between each handler and ending the sequence with `H0`. In addition to the name of the service, `Server` has an additional parameter which records the types of the handlers. Since that list may become quite long, we can ask GHC to write it for us by using the `PartialTypeSignatures` extension and writing an underscore `_` in that position.
```

Comment:
```suggestion
Since you can declare more than one method in a service, you need to join them into a `Server`. You do so by using `(:<|>:)` between each handler and ending the sequence with `H0`. In addition to the name of the service, `Server` has an additional parameter which records the types of the handlers. Since that list may become quite long, we can ask GHC to write it for us by using the `PartialTypeSignatures` extension and writing an underscore `_` in that position.
```